Rendering/Effects/OffGeometryEffect.cpp

```diff
@@ -68,21 +72,10 @@ void OffGeometryEffect::changeOffset(const buw::Vector3d& offsetOld, const buw::
 		{
 			std::vector<buw::Vector3d> vertices;
 			vertices.reserve(offModel_->geometry().vertices.size());
-			for (auto& vertex : offModel_->geometry().vertices)
-			{
-				buw::Vector3d vtx(vertex);
-				vtx[0] += offsetNew.x();
-				vtx[1] += offsetNew.y();
-				vtx[2] += offsetNew.z();
-				vertices.push_back(vtx);
-			}
-
-			const uint32_t verticesOffset = verticesMesh.size();
-			verticesMesh.insert(verticesMesh.end(), vertices.begin(), vertices.end());
-
-			const uint32_t indicesOffset = indicesMesh.size();
-			indicesMesh.insert(indicesMesh.end(), offModel_->geometry().indices.begin(), offModel_->geometry().indices.end());
-			std::for_each(indicesMesh.begin() + indicesOffset, indicesMesh.end(), [&verticesOffset](uint32_t& index) {index += verticesOffset; });
+			std::transform(offModel_->geometry().vertices.begin(),
+				offModel_->geometry().vertices.end(),
+				vertices.begin(),
+				[offsetNew](buw::Vector3d a) {a.x() += offsetNew.x(); a.y() += offsetNew.y(); a.z() += offsetNew.z(); return a; });
```

Comment:
I believe `buw::vector3d` have addition defined, so one could `return a + offsetNew;`.

If it's not possible, then ignore the comment. (although - how come it is not possible?!)

Reply:
Makes sense, I will try.

Reply:
Resolved: 586ab1f